build(deps): upgrade vulnerable dependencies to fix security issues

mrkimani-ndegwa opened the pull request:
Upgrade multiple dependencies to address security vulnerabilities:

- word-wrap: 1.2.3 → 1.2.5
- semver: 6.3.0 → 6.3.1
- decode-uri-component: 0.2.0 → 0.2.2
- shelljs: 0.8.4 → 0.8.5

These upgrades fix 89 security vulnerabilities:
- 81 High-severity vulnerabilities
- 8 Moderate-severity vulnerabilities

Total vulnerabilities reduced from 274 to 185.

Added yarn resolutions to force upgrade of transitive dependencies. Verified all packages build successfully.

Addresses security PRs: #126, #125, #122, #112, #111, #109, #108, #107, #105

Commit:
build(deps): upgrade vulnerable dependencies to fix security issues

Upgrade multiple dependencies to address security vulnerabilities:

- word-wrap: 1.2.3 → 1.2.5
- semver: 6.3.0 → 6.3.1
- decode-uri-component: 0.2.0 → 0.2.2
- shelljs: 0.8.4 → 0.8.5

These upgrades fix 89 security vulnerabilities:
- 81 High-severity vulnerabilities
- 8 Moderate-severity vulnerabilities

Total vulnerabilities reduced from 274 to 185.

Added yarn resolutions to force upgrade of transitive dependencies.
Verified all packages build successfully.

Addresses security PRs: #126, #125, #122, #112, #111, #109, #108, #107, #105

Seismic-Security-Service commented:
### :no_entry: **Snyk checks have failed. 1 issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (1) |
|---|---|---|---|---|---|---|
| :no_entry: | Open Source Security| 0 | 1 | 0 | 0 | [ 1 issues](https://app.snyk.io/org/percolate-platform/pr-checks/323f2498-1106-49a6-92b7-04e17e58ee75?refs=7566f6617374496fa1adb96ab5a35d1b569404ad&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/percolate-platform/pr-checks/323f2498-1106-49a6-92b7-04e17e58ee75/license?refs=7566f6617374496fa1adb96ab5a35d1b569404ad&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

Commit:
Add missing vulnerability resolutions from Snyk report

Added resolutions for 10 additional security vulnerabilities:
- ansi-regex: ^5.0.1
- @babel/traverse: ^7.23.2
- istanbul-reports: ^3.1.3
- nwsapi: ^2.2.1
- prompts: ^2.4.2
- tough-cookie: ^4.1.3
- ws: ^7.5.10
- braces: ^3.0.3
- micromatch: ^4.0.6
- minimist: ^1.2.6

Completes Snyk security scan findings per Seismic standard.
